types/hooks.d.ts

```diff
@@ -498,7 +498,7 @@ export type onTimeoutMetaHookHandler<
 /**
  * This hook is useful if you need to do some custom error logging or add some specific header in case of error.
  * It is not intended for changing the error, and calling reply.send will throw an exception.
- * This hook will be executed only after the customErrorHandler has been executed, and only if the customErrorHandler sends an error back to the user (Note that the default customErrorHandler always sends the error back to the user).
+ * This hook will be executed only before the customErrorHandler is executed. The onError hook is useful for logging errors or adding custom headers, but it cannot modify the error or prevent the error handler from running (Note that the default customErrorHandler always sends the error back to the user).
```

Comment:
```suggestion
 * This hook will be executed before the customErrorHandler.
```

The purpose of this hook is already explained above.